### ISSUE TITLE
[NEWS] Dodaliśmy nową strukturę w modelu oferty - **Pasuje do pojazdów**

### ISSUE THREAD

**Issue by KrzysztofLukanowski** [allegro/allegro-api#477]
Dodaliśmy nową strukturę w modelu oferty - **Pasuje do pojazdów**
W metodach:

- POST na zasobie /sale/offers
- PUT na zasobie /sale/offers/{offerId}
- GET na zasobie /sale/offers/{offerId}

wdrożyliśmy nową sekcję. Jeśli sprzedający oferuje przedmiotu w jednej z kategorii Części samochodowych:

- [Układ chłodzenia silnika](https://allegro.pl/kategoria/czesci-samochodowe-uklad-chlodzenia-silnika-18689?order=m)
- [Układ hamulcowy](https://allegro.pl/kategoria/czesci-samochodowe-uklad-hamulcowy-18834?order=m)
- [Układ kierowniczy](https://allegro.pl/kategoria/czesci-samochodowe-uklad-kierowniczy-250842?order=m)
- [Układ napędowy](https://allegro.pl/kategoria/czesci-samochodowe-uklad-napedowy-50863?order=m)
- [Układ wydechowy](https://allegro.pl/kategoria/czesci-samochodowe-uklad-wydechowy-18862?order=m)

W tej sekcji **dodaj listę marek i modeli aut,** do których pasuje oferowana część. Sekcję wyświetlimy pod opisem oferty. 

Przeczytaj więcej w artykule na portalu [REST API Allegro](https://developer.allegro.pl/news/2018-08-08-Pasuje/).

**Comment by zulus** [comment]
Nareszcie choć szkoda że nie tecdoc :P 

A co z innymi kategoriami w ramach motoryzacji? Filtry, oświetlenie, wycieraczki? Część informacji ma lądować w standardowym opisie a część w dodatkowych strukturach??

Największy problem to "części" typu pióra wycieraczek / żarówki. Tam jest kluczowe które auto pasuje i zwykle nie mieści się w standardowym opisie.

**Comment by KrzysztofLukanowski** [comment]
@zulus to jest start nowej opcji, badamy zapotrzebowanie w kolejnych kategoriach. Będziemy informować jak tylko poszerzymy zakres kategorii, w której można z niej skorzystać.

**Comment by kfijalko** [comment]
TecDoc w ramch opłaty licencyjnej pobiera procent od obrotu :)
To by oznaczało wzrost prowizji od sprzedaży w tych kategoriach - JA NIE CHCĘ TECDOC'a !
Poza tym Tecdoc nie sprawdza się w częściach karoseryjnych i spowodował by duży problem.

Baza tecdoc to baza tworzona przez producentów zamienników a nie producentów samochodów.
Znakomita większość zamieników w nadwoziach produkowana jest w Chinach i na Tajwanie, i nie ma ich w tecdoc.
Największe polskie hurtownie tworzą osobne katalogi dla mechaniki i karoserii. Ze względu na to, że nie da się łatwo przemapować słowników modeli mechnicznych na słownik modeli nadwoziowych.
20 lat temu była taka tendencja, że przez 6-10 lat szło jedno nadwozie i w między czasie zmieniała się dwa/trzy razy budowa mechaniczna/termiczna. 
Teraz tendencja jest taka, że 6-10 lat idzie jedna mechanika/termika a co dwa lata jest lifting nadwozia.


Sam pomysł wprowadzenia zakładki zastosowania, która będzie mogła w przyszłości zamienić się w filtry jest bardzo dobra. Ale trzeba to zrobić bardzo rozsądnie, koniecznie z konsultacjami ze sprzedawcami części.

Pozdrawiam,kf

**Comment by zulus** [comment]
@kfijalko w przypadku tecdoc chodziło mi bardziej o jakieś zesłownikowanie tego. My zwykle podajemy moc silnika w KM i lata produkcji, allegro jako przykład daje KM/ kW i ccm inny postanowi dać to jeszcze inaczej przez co niemożliwością stanie się jednoznaczne pokazanie części z danej kategorii pasujące do konkretnego auta. Zwłaszcza jeśli allegro miało by się kiedyś dorobić wyszukiwania po VIN. W przypadku części spoza teca, budujemy sobie wewnętrznie bazę zgodną z ich modelem, w szczególności jeśli chodzi o listę aut (typnr).

**Comment by zulus** [comment]
@KrzysztofLukanowski chciałem się upewnić, czy zawartość tych pól jest przeszukiwana po zaznaczeniu wyszukiwania w opisach ofert?

**Comment by kfijalko** [comment]
@zulus czy badaliście temat rozkodowywania VIN w sensie kupienia gdzieś bazy VIN->marka->model->silnik->cżęści->numery OE. Podobno któraś hurtownia udostępniła wyszukiwanie częsci po VIN, ale mi się nie udało znaleźć takiej bazy. Dwa lata temu podszedłem do cepika, ale jak zajzałem do bazy to poległem przy próbie mapowania VIN->marka->model->silnik. My się bawimy w ręczne rozkodowanie w "różnych serwisach/programach". Chciałbym zbudować coś automatycznego.

**Comment by KrzysztofLukanowski** [comment]
@zulus klient nie będzie musiał zaznaczyć szukania wewnątrz opisów, bez tej opcji sprawdzimy zawartość tych pól i zwrócimy listę ofert, która będzie jak najlepiej odpowiadać jego zapytania. Problem w tym teraz, że aktualnie mamy małe niedogodności z indeksowaniem zawartości tych pól. W każdym razie pracujemy nad tym, aby wszystko działało tak jak powinno.

**Comment by zulus** [comment]
@kfijalko napisz na mój e-mail w profilu to Ci odpowiem
@KrzysztofLukanowski idealnie, praca z ElasticSearch bywa przytłaczająca na początku ;)

**Comment by jakubmaguza** [comment]
Nie wiem jak to wygląda licencyjnie, ale nie możecie po prostu użyć bazy MVL?
https://pages.ebay.co.uk/help/sell/contextual/master-vehicle-list-manually.html?
Te numerki pasują do TecDoc. Na pewno łatwiej jest dodawać samochody po numerze niż po opisie.
Pewnie też dla was łatwiej to dekodować.

Oczywiście nie jest to dokładne, bo w ramach jednego kodu pojazdu może być kilka wersji.

**Comment by skozuchowicz** [comment]
@jakubmaguza , można MVL obrobić sobie w swoim interfejsie i ostatecznie wysyłać string do Allegro.

**Comment by jakubmaguza** [comment]
Oczywiście, to nie jest najmniejszy problem. Bardziej chodzi mi o powód wybrania stringów zamiast kodów liczbowych. Prawdopodobnie jak ktoś wspomniał wyżej, to kwestie licencyjne.

**Comment by KrzysztofLukanowski** [comment]
Mogę zdradzić, że pracujemy już nad rozwiązaniem z zamkniętą listą słownikową pojazdów (baza pojazdów). Jeżeli macie jakieś propozycje odnośnie kształtu API, dajcie znać. Bierzemy pod uwagę każdy głos.
